libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1109 @@
+    } else {
+        fc_status.bits.aircraft_type = static_cast<uint8_t>(AircraftType::Aircraft_Vtol);
+    }
+    
+    float rel_height;
+    AP::ahrs().get_relative_position_D_home(rel_height);
+
+    // fill in packet
+    MessageAhrs msg_ahrs = {
+        .gps_update_flag = _gps_update_count,
+        .fc_status = {fc_status.all},
+        .fc_gyro = {gyro_NED.x, gyro_NED.y, gyro_NED.z},
+        .fc_acc = {acc_NED.x,acc_NED.y,acc_NED.z},
+        .vel = {vel_NED.x, vel_NED.y, vel_NED.z},
+        .euler = {degrees(AP::ahrs().get_pitch()), degrees(AP::ahrs().get_roll()), degrees(AP::ahrs().get_yaw())},
+        .relative_height = rel_height,
+        .altitude = alt_amsl_cm*0.01,
+        .longitude = loc.lng*1e-7,
+        .latitude = loc.lat*1e-7,
+    };
+
+    FrameType frame_to_send{
+        .flag={0},
+        data_len : sizeof(msg_ahrs),
+        cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+        sub_cmd : static_cast<uint8_t>(GimbalSubCmd::GimbalSubCmd_PlaneSensor_Attitude),
+        data : (uint8_t*)&msg_ahrs
+    };
+
+    
+    // send packet to gimbal
+    return send_packet(frame_to_send);
+}
+
+// take a picture.  returns true on success
+bool AP_Mount_Tusuav::take_picture()
+{
+    // exit immediately if not initialised
+    if (!_initialised) {
+        return false;
+    }
+
+    RecordPhotoSettings settings = {.bits={1,RecordingCmd::No_Action}};
+    FrameType frame_to_send {
+        .flag={0},
+        .data_len = sizeof(settings),
+        .cmd = static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+        .sub_cmd = static_cast<uint8_t>(GimbalSubCmd::gimbal_single_cmd_record_switch),
+        .data = (uint8_t*)&settings
+    };
+    return send_packet(frame_to_send);
+}
+
+// start or stop video recording.  returns true on success
+// set start_recording = true to start record, false to stop recording
+bool AP_Mount_Tusuav::record_video(bool start_recording)
+{
+    debug("recording %d", _initialised);
+    // exit immediately if not initialised
+    if (!_initialised) {
+        return false;
+    }
+    RecordPhotoSettings settings = {.bits={0,RecordingCmd::No_Action}};
+
+    if (start_recording) {
+        settings.bits.record = RecordingCmd::Start;
+    } else {
+        settings.bits.record = RecordingCmd::Stop;
+    }
+
+    FrameType frame_to_send{
+    .flag={0},
+    data_len : sizeof(settings),
+    cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+    sub_cmd : static_cast<uint8_t>(GimbalSubCmd::gimbal_single_cmd_record_switch),
+    data : (uint8_t*)&settings
+    };
+
+    // send packet to gimbal
+    return send_packet(frame_to_send);
+
+}
+
+// set zoom specified as a rate or percentage
+bool AP_Mount_Tusuav::set_zoom(ZoomType zoom_type, float zoom_value)
+{
+    // exit immediately if not initialised
+    if (!_initialised) {
+        return false;
+    }
+
+    int32_t int_zoom_value = (int32_t)zoom_value;
```

Comment:
```suggestion
    const int32_t int_zoom_value = (int32_t)zoom_value;
```

... and in general just const everything that can be in here

Reply:
Thank you for the suggestion. We’ve updated the code to apply `const` where applicable.

Reply:
Thank you for the suggestion. We’ve updated the code to apply 'const' where applicable.